src/constraint.rs

```diff
@@ -82,16 +82,25 @@ impl Constraint {
 
     /// Creates a [Constraint::ContainerLength] from a [Range] specifying a length range.
     pub fn container_length(length: Range) -> Constraint {
+        if !matches!(length, Range::IntegerNonNegative(_, _)) {
+            panic!("container_length constraint  must have a range of type IntegerNonNegative")
+        }
         Constraint::ContainerLength(ContainerLengthConstraint::new(length))
     }
 
     /// Creates a [Constraint::ByteLength] from a [Range] specifying a length range.
     pub fn byte_length(length: Range) -> Constraint {
+        if !matches!(length, Range::IntegerNonNegative(_, _)) {
+            panic!("byte_length constraint  must have a range of type IntegerNonNegative")
```

Comment:
```suggestion
            panic!("byte_length constraint must have a range of type IntegerNonNegative")
```

I see this was copy/pasted for each function, so I'll refrain from commenting on each one.